libs-haskell/da-hs-base/src/DA/Service/Logger/Impl/GCP.hs

```diff
@@ -158,14 +161,26 @@ validateSentData = \case
       then pure sentData
       else noSentData
 
-initialiseGcpState :: GCPConfig -> Lgr.Handle IO -> IO GCPState
-initialiseGcpState GCPConfig{..} gcpFallbackLogger = do
+initialiseGcpState :: GCPConfig -> Lgr.Handle IO -> IO (Maybe GCPState)
+initialiseGcpState GCPConfig {..} gcpFallbackLogger = do
+    gcpCacheDirM <- getCacheDir gcpConfigCachePath
+    -- copy machine_id file to cache directory to preserve statistics.
+    let damlPathAndCachePathM = do
+            cachePath <- gcpCacheDirM
+            damlPath <- gcpConfigDamlPath
+            Just (cachePath, damlPath)
+    forM_ damlPathAndCachePathM $ \(cachePath, damlPath) -> do
+        let machineIdPath = damlPath </> ".machine_id"
+        let newMachineIdPath = cachePath </> ".machine_id"
+        hasMachineId <- doesFileExist machineIdPath
+        when hasMachineId $ do
+          copyFile machineIdPath newMachineIdPath
```

Comment:
This should also have a check that `newMachineIdPath` does not yet exist. We don’t want to overwrite an already existing file. It’s probably rarely an issue but better be safe than sorry.

Reply:
I added the check, that's a good idea. I'll also test locally before merging.